src/components/hooks/use-tab-state.js

```diff
@@ -0,0 +1,32 @@
+import { useState, useEffect } from 'react'
+import { checkElemInArray, getLocationHash, isBrowser, routeBack, scrollTop, setLocationHash,  } from 'common/utility'
+
+export const useTabState = (tab_list) => {
+    const [active_tab, setActiveTab] = useState(getLocationHash() ? getLocationHash() : tab_list[0])
```

Comment:
we only rely on hash when its mounted, i guess we can add getlocation hash inside didmount useEffect. and then listen to active_tab to set location hash scrolltop, or routeback

Reply:
This sets the location if the user intendedly enter the URL with the hash provided.